src/main/java/com/esri/arcgisruntime/toolkit/skins/FloorFilterSkin.java

```diff
@@ -0,0 +1,751 @@
+import javafx.scene.control.Label;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.SkinBase;
+import javafx.scene.control.TextField;
+import javafx.scene.control.TitledPane;
+import javafx.scene.layout.VBox;
+
+import static javafx.scene.input.MouseEvent.MOUSE_CLICKED;
+
+/**
+ * <p>A skin for displaying a {@link FloorFilter}.</p>
+ *
+ * <p>The skin is only configured and displayed if there is a valid FloorManager loaded in the data model.</p>
+ *
+ * <p>FloorSite and FloorFacility selection occurs within titled panes containing list views of the sites and facilities
+ * attached to the floor manager. FloorLevels are displayed in a list view when a facility that has levels is selected.</p>
+ *
+ * <p>If there is only 1 site on the floor manager, it is automatically selected and the sites browser will not be
+ * displayed.</p>
+ *
+ * <p>Styles can be customized using the set style classes. The default style class set in the Control is
+ * "floor-filter-view".>/p>
+ *
+ * @since 100.14.0
+ */
+public class FloorFilterSkin extends SkinBase<FloorFilter> {
+
+  // used to manage list view content
+  private static final double MAX_NO_OF_ROWS = 5.0;
+  private static final double CELL_SIZE = 26.0;
+
+  private static final double PREF_WIDTH = 220.0;
+
+  private final FloorFilter skinnable;
+
+  // main panes and scroll pane to manage content if height overruns
+  private final VBox floorFilterPane = new VBox();
+  private final ScrollPane scrollPane = new ScrollPane();
+  private final VBox contentPane = new VBox();
+
+  // sites and facilities section
+  private final VBox sitesAndFacilitiesVBox = new VBox();
+
+  private final TitledPane sitesTitledPane = new TitledPane();
+  private final VBox sitesVBox = new VBox();
+  private final Label sitesHeading = new Label("Select a site");
+  private final TextField sitesFilterTextField = new TextField();
+  private final ListView<FloorSite> sitesListView = new ListView<>();
+  private final CheckBox allSitesCheckbox = new CheckBox("All sites");
+
+  private final TitledPane facilitiesTitledPane = new TitledPane();
+  private final VBox facilitiesVBox = new VBox();
+  private final Label facilityHeading = new Label("Select a facility");
+  private final TextField facilitiesFilterTextField = new TextField();
+  private final ListView<FloorFacility> facilitiesListView = new ListView<>();
+
+  // levels section
+  private final VBox levelsVBox = new VBox();
+  private final Label levelsHeading = new Label("Levels:");
+  private final ListView<FloorLevel> levelsListView = new ListView<>();
+  private final CheckBox allLevelsCheckbox = new CheckBox("Show all levels");
+
+  // boolean properties for controlling visibility of sections
+  // property to toggle the visibility of the sites titled pane
+  private final SimpleBooleanProperty isShowSitesProperty = new SimpleBooleanProperty();
```

Comment:
The alternative would be `isSitesVisible` but I think it's maybe then implying the sites layer isn't visible given the data in this particular tool - I've commented each property specifically to explain it's for the visibility of the UI component, so I think I'll stick to avoid going to a really long variable name.

Reply:
It should be called `showSitesProperty`. As an example see https://openjfx.io/javadoc/11/javafx.graphics/javafx/scene/Node.html#visibleProperty() where you have `visibleProperty` for the property by `isVisible` for the getter.

Reply:
Thanks that makes sense. I'll go through the Boolean properties 👍